cargo-pgrx/src/command/init.rs

```diff
@@ -275,6 +275,77 @@ fn untar(bytes: &[u8], pgrxdir: &PathBuf, pg_config: &PgConfig) -> eyre::Result<
     }
 }
 
+fn fixup_homebrew_for_icu(configure_cmd: &mut Command) {
+    // See if it's disabled via an argument
+    if configure_cmd.get_args().any(|a| a == "--without-icu") {
+        return;
+    }
+    // Check if the user has manually configured the vars that `./configure`
+    // understands.
+    // (ideally these vars should be read off of `configure_cmd.get_envs()`
+    // before falling back to the system env, but this is involved enough)
+    let icu_cflags = std::env::var_os("ICU_CFLAGS").is_some();
+    let icu_libs = std::env::var_os("ICU_LIBS").is_some();
+    if icu_cflags || icu_libs {
+        return;
+    }
+    // Check if pkg-config knows where they are. (note: `$PKG_CONFIG` is a var
+    // respected by `./configure`)
+    let pkg_config = std::env::var_os("PKG_CONFIG").unwrap_or_else(|| "pkg-config".into());
+    let exists_result =
+        Command::new(pkg_config).arg("--exists").arg("\"icu-uc icu-i18n\"").output();
+    if exists_result.map(|out| out.status.success()).unwrap_or_default() {
+        // pkg-config knows where they are already, so we don't have to do
+        // anything.
+        return;
+    }
+    // See if they use homebrew, and have `icu4c` installed.
+    let res = Command::new("brew").arg("--prefix").arg("icu4c").output();
+    let Ok(output) = res else {
+        // They aren't a brew user (or `brew` errored for some other reason), so
+        // we can't help them.
+        return;
+    };
+    let icu4c_pkgconfig_path =
+        PathBuf::from(String::from_utf8_lossy(&output.stdout).trim()).join("lib").join("pkgconfig");
+
+    if !output.status.success()
+        || !icu4c_pkgconfig_path.exists()
+        || !icu4c_pkgconfig_path.is_absolute()
+    {
+        let msg = "\
+            Homebrew seems to be in use, but the `icu4c` does not appear to be \
+            installed. Without this, the build of PostgreSQL 16 may fail. You \
+            may be able to fix this by installing it using:\n\
+            \n\
+            $ brew install icu4c\n\
+            \n\
+            before retrying. Alternatively, if you are certain you do not need it \
+            you may disable ICU support during the build by invoking `cargo pgrx init` \
+            with `--configure-flags=--without-icu`, as follows:\n\
```

Comment:
... but the `icu4c` package is not installed.

Reply:
I think it just controls collations, if their extension doesn't care about that ISTM like it's reasonable to disable ICU.